### PR TITLE
Remove item and sprite images from runelite-client resources directory

### PR DESCRIPTION
**Pull request by Nightfirecat**
Many plugins made use of image files which were exact copies (or close derivatives) of ingame item or sprite images. As these images can be generated using `ItemManager` and `SpriteManager`, this PR seeks to remove as many of these extraeneous images as possible.

After these commits, the images that remain are non-duplicated custom images with the following exceptions:

* resources/net/runelite/client/plugins/cluescrolls/**emote.png**; this is a close copy of the interface styles' 2010 emote menu tab icon, but is scaled differently and is centered within the image canvas
* resources/net/runelite/client/plugins/farmingtracker/**farming.png**; this is distinct from any existing farming icons I could find, and does not seem to be a simple resize of other farming icons
* resources/net/runelite/client/plugins/**poh/\*** and resources/net/runelite/client/plugins/**worldmap/\***; these images are all of different dimensions and makeups from spell icons, and seem to be created specially for the minimap and world map, so have been left unaltered
* **skill_icons/\*** and **skill_icons_small/\***; skill_icons_small are copies of the images found on the online highscore webpage, and are not present in the client cache. skill_icons are found in the cache, but must be consumed by the highscores plugin which loads even when the client is outdated and cannot generate a `SpriteManager` to load from cache. Hence, both directories are necessary.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityArenaTimer.java [hunk 1]
```diff
@@ -25,41 +25,31 @@
...
+	AgilityArenaTimer(Plugin plugin, BufferedImage image)
 	{
-		super(1, ChronoUnit.MINUTES, getTicketImage(), plugin);
+		super(1, ChronoUnit.MINUTES, image, plugin);
 		setTooltip("Time left until location changes");
 	}
 
 	private static BufferedImage image;
-	private static BufferedImage getTicketImage()
+	static BufferedImage getTicketImage(final ItemManager itemManager)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityArenaTimer.java]: This should probably be in the AgilityPlugin

File: runelite-client/src/main/java/net/runelite/client/game/ClanManager.java [hunk 2]
```diff
@@ -231,4 +225,22 @@ private static BufferedImage rgbaToIndexedBufferedImage(final BufferedImage sour
 		resultIndexedImage.getGraphics().drawImage(sourceBufferedImage, 0, 0, null);
 		return resultIndexedImage;
 	}
+
+	private void setClanChatImages()
+	{
+		int i = 0;
+		for (int resource : CLANCHAT_IMAGES)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/game/ClanManager.java]: Just use normal for() loop when you also need index

File: runelite-client/src/main/java/net/runelite/client/game/ClanManager.java [hunk 3]
```diff
@@ -95,26 +106,11 @@ public ClanMemberRank load(String key) throws Exception
 		});
 
 	@Inject
-	public ClanManager(Provider<Client> clientProvider)
+	public ClanManager(Client client, ClientThread clientThread, SpriteManager spriteManager)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/game/ClanManager.java]: Make this constructor private

File: runelite-client/src/main/java/net/runelite/client/game/ClanManager.java [hunk 4]
```diff
@@ -37,36 +39,46 @@
...
+			SpriteID.CLAN_CHAT_RANK_CROWN_JAGEX_MODERATOR,
 		};
+	private static final Dimension CLANCHAT_IMAGE_DIMENSION = new Dimension(11, 11);
+	private static final Color CLANCHAT_IMAGE_OUTLINE_COLOR = new Color(33, 33, 33);
 
+	private Boolean clanChatImagesSet = false;
 	private int modIconsLength;
 
-	private final Provider<Client> clientProvider;
+	private final Client client;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/game/ClanManager.java]: Final fields should be above non-final fields

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollWorldMapPoint.java [hunk 5]
```diff
@@ -32,24 +32,23 @@
 
 class ClueScrollWorldMapPoint extends WorldMapPoint
 {
-	private static final BufferedImage CLUE_SCROLL_WORLD_IMAGE;
-	private static final Point CLUE_SCROLL_WORLD_IMAGE_POINT;
+	private final ClueScrollPlugin plugin;
+	private final BufferedImage CLUE_SCROLL_WORLD_IMAGE;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollWorldMapPoint.java]: As this is not constant anmore, fix naming

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollWorldMapPoint.java [hunk 6]
```diff
@@ -32,24 +32,23 @@
...
+	private final Point CLUE_SCROLL_WORLD_IMAGE_POINT;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollWorldMapPoint.java]: same here

File: runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathWorldMapPoint.java [hunk 7]
```diff
@@ -26,45 +26,29 @@
...
 import net.runelite.api.coords.WorldPoint;
-import static net.runelite.client.plugins.deathindicator.DeathIndicatorPlugin.BONES;
 import net.runelite.client.ui.overlay.worldmap.WorldMapPoint;
 
 class DeathWorldMapPoint extends WorldMapPoint
 {
-	private static final BufferedImage WORLDMAP_HINT_ARROW;
-	private static final Point WORLDMAP_HINT_ARROW_POINT;
+	private final DeathIndicatorPlugin plugin;
+	private final BufferedImage WORLDMAP_HINT_ARROW;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathWorldMapPoint.java]: Naming

File: runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathWorldMapPoint.java [hunk 8]
```diff
@@ -26,45 +26,29 @@
...
+	private final Point WORLDMAP_HINT_ARROW_POINT;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathWorldMapPoint.java]: Naming

File: runelite-client/src/main/java/net/runelite/client/util/SwingUtil.java [hunk 9]
```diff
@@ -307,6 +310,103 @@ private static BufferedImage resizeImage(BufferedImage image, int newWidth, int
...
 
+	/**
+	 * Re-size a BufferedImage's canvas to the given dimensions.
+	 *
+	 * @param image     The image whose canvas should be re-sized.
+	 * @param newWidth  The width to set the BufferedImage to.
+	 * @param newHeight The height to set the BufferedImage to.
+	 * @return          The BufferedImage centered within canvas of given dimensions.
+	 */
+	public static BufferedImage resizeCanvas(final BufferedImage image, final int newWidth, final int newHeight)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/util/SwingUtil.java]: This is a lot of new Image-related methods. maybe time for ImageUtil?

File: runelite-client/src/main/java/net/runelite/client/util/SwingUtil.java [hunk 10]
```diff
@@ -395,4 +495,55 @@ public static Color colorLerp(Color a, Color b, double t)
...
+					continue;
+				}
+
+				filledImage.setRGB(x, y, color.getRGB());
+			}
+		}
+		return filledImage;
+	}
+
+	private static boolean isFullyTransparent(final Color color)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/util/SwingUtil.java]: These 2 below methods seem unnecessary

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/util/SwingUtil.java]: These were written so I wouldn't have to use unnamed default values in the `outlineImage` methods. I can see removing `isFullyTransparent` and moving its logic into `isNotFullyTransparent`, but I'm not sure of a more graceful way to set those default values.

**Comment by Nightfirecat** [comment]
Came across `SwingUtil::grayscaleOffset`, so I'm pretty sure I can remove a couple more images with that. Will comment once this is ready for review again.

**Comment by Nightfirecat** [comment]
Don't want this merged just yet, will re-open when it's ready.

**Pull request closed without merging**